utils.py

```diff
@@ -1022,30 +1022,40 @@ def get_fake_context(context=None, area_type='VIEW_3D'):
 
 
 def has_url(text):
-    # urls = re.findall('https?://(?:[-\w.]|(?:%[\da-fA-F]{2}))+', text)
-    urls = re.findall(r'(?<=<a href=")[^"]*', text)
-    urls_with_str = []
+    #first remove markdown *
+    text = text.replace('*','')
+    # Anything that isn't a square closing bracket
+    name_regex = "[^]]+"
+    # http:// or https:// followed by anything but a closing paren
+    url_regex = "http[s]?://[^)]+"
+    markup_regex = '\[({0})]\(\s*({1})\s*\)'.format(name_regex, url_regex)
```

Comment:
I would suggest to use f-strings as our default first choice in the add-on.

`markup_regex = f"\[({name_regex})]\(\s*({url_regex})\s*\)"`